sos/report/plugins/gluster.py

```diff
@@ -41,15 +41,14 @@ def wait_for_statedump(self, name_dir):
                     ret = string.count(last_line, 'DUMP_END_TIME')
 
     def postproc(self):
-        if not os.path.exists(self.statedump_dir):
-            return
-        try:
-            for dirs in os.listdir(self.statedump_dir):
-                os.remove(os.path.join(self.statedump_dir, dirs))
-            os.rmdir(self.statedump_dir)
-            os.unlink('/tmp/glusterdump.options')
-        except OSError:
-            pass
+        if self.get_option("dump"):
+            if not os.path.exists(self.statedump_dir):
+                return
+            try:
+                for name in glob.glob(self.statedump_dir + '/*.dump.[0-9]*'):
```

Comment:
not sure if I'm missing something, don't we need to have '.' after the '/'?

Reply:
Threw me off at first too, but `glob` uses UNIX-style regex, not python-style.

Reply:
No - these are not "dotfiles". They start with a string like "glusterdump" or a mount point and end with  ".dump.$TIMESTAMP":

```# ls /var/run/gluster/*.dump.[0-9]*
/var/run/gluster/glusterdump.1452.dump.1594652912   /var/run/gluster/glusterdump.24959.dump.1594652913  /var/run/gluster/glusterdump.8205.dump.1594813835   /var/run/gluster/mnt-data2-2.24719.dump.1594652912
/var/run/gluster/glusterdump.1452.dump.1594652913   /var/run/gluster/glusterdump.24959.dump.1594813835  /var/run/gluster/glusterdump.8245.dump.1594652912   /var/run/gluster/mnt-data2-2.24719.dump.1594652913
/var/run/gluster/glusterdump.1452.dump.1594813835   /var/run/gluster/glusterdump.24961.dump.1594652912  /var/run/gluster/glusterdump.8245.dump.1594652913   /var/run/gluster/mnt-data2-2.24719.dump.1594813835
/var/run/gluster/glusterdump.1454.dump.1594652912   /var/run/gluster/glusterdump.24961.dump.1594652913  /var/run/gluster/glusterdump.8245.dump.1594813835   /var/run/gluster/var-lib-glusterd-ss_brick.14293.dump.1594652912
/var/run/gluster/glusterdump.1454.dump.1594652913   /var/run/gluster/glusterdump.24961.dump.1594813835  /var/run/gluster/mnt-data1-1.24697.dump.1594652912  /var/run/gluster/var-lib-glusterd-ss_brick.14293.dump.1594652913
/var/run/gluster/glusterdump.1454.dump.1594813835   /var/run/gluster/glusterdump.8205.dump.1594652912   /var/run/gluster/mnt-data1-1.24697.dump.1594652913  /var/run/gluster/var-lib-glusterd-ss_brick.14293.dump.1594813835
/var/run/gluster/glusterdump.24959.dump.1594652912  /var/run/gluster/glusterdump.8205.dump.1594652913   /var/run/gluster/mnt-data1-1.24697.dump.1594813835
```

Reply:
No, glob uses ["globs"](https://en.wikipedia.org/wiki/Glob_(programming)) :-D - they are a totally different (although superficially similar) string pattern matching scheme that originates in the Bourne shell.

Reply:
I don't have knowledge of how glob works. if it is something not missed by mistake, I'm good.